src/internal/m365/collection/site/backup.go

```diff
@@ -188,6 +189,20 @@ func CollectLists(
 
 		storageDir := path.Elements{listID}
 
+		var (
+			dp          = dps[storageDir.String()]
+			prevPathStr = dp.Path
+		)
+
+		if len(prevPathStr) > 0 {
+			if prevPath, err = pathFromPrevString(prevPathStr); err != nil {
```

Comment:
We should add a test for previous paths, or metadata collections in general. This can be at `ProduceBackupCollection` level. You can use `TestCreateGroupsCollection_SharePoint_InvalidMetadata` as a reference. 

Please ignore if your upcoming PRs already do this.

Reply:
yes I believe testing of this part is taken care in the last PR in the chain